controller/comments.go

```diff
@@ -130,15 +133,13 @@ func (c *CommentsController) performUpdate(ctx *app.UpdateCommentsContext, cm *c
 		cm.Markup = rendering.NilSafeGetMarkup(ctx.Payload.Data.Attributes.Markup)
 		err := appl.Comments().Save(ctx.Context, cm, *identityID)
 		if err != nil {
-			return jsonapi.JSONErrorResponse(ctx, err)
+			return err
```

Comment:
@xcoulon In the other changes that you've made, you did capture the error returned from `application.Transaction` and called `jsonapi.JSONErrorResponse` on it. Here you don't do that but I assume that the result of the call to `performUpdate` (the housing function) is converted to a JSONAPI error. Will check that.

Reply:
Hm, the call to `performUpdate` is not in your code so I've checked the [`controller/comments.go`](https://github.com/fabric8-services/fabric8-wit/blob/9f667e5880b0941d054166a9d14fdff1887528a4/controller/comments.go#L120) in master and found that it is being called only once like this:

```go
// Update does PATCH comment
func (c *CommentsController) Update(ctx *app.UpdateCommentsContext) error {

	// .... SKIPPED OVER SOME CODE ....

	result := c.performUpdate(ctx, cm, identityID)
	if ctx.ResponseData.Status == 200 {
		c.notification.Send(ctx, notification.NewCommentUpdated(cm.ID.String()))
	}
	return result
}
```

To me this looks wrong because 

1. the error is obviously not converted to a JSONAPI error here and
2. the notification is sent nomatter what the error (here `result`) was.

@aslakknutsen what do you think?